preprocessing/neighbors/n_neighs.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+
+# Author_and_contribution: Niklas Mueller-Boetticher; created script, 
+# Author_and_contribution: Qirong Mao; implemented method
+
+import argparse
+
+# TODO adjust description
+parser = argparse.ArgumentParser(
+    description="Constructing spatial neighborhood with custom parameters"
+)
+
+parser.add_argument(
+    "-c", "--coordinates", help="Path to coordinates (as tsv).", required=True
+)
+
+parser.add_argument(
+    "-m", "--matrix", help="Path to (transformed) counts (as mtx).", required=True
+)
+
+parser.add_argument(
+    "-f", "--features", help="Path to features (as tsv).", required=True
+)
+
+parser.add_argument(
+    "-o", "--observations", help="Path to observations (as tsv).", required=True
+)
+
+parser.add_argument(
+    "--coord_type", default=None, choices=['grid', 'generic', None],help="Type of coordinate system.", required=False
```

Comment:
I think n_neigs should only be used for non-grid data as the definition doies not really make sense for grid data. How do you have e.g. 10 neighbors in a hexagonal grid

Reply:
But also possible with ~19 neighbors (two layers), but it can be replaceable by just setting `n_rings`

Reply:
But n_rings is already handled by a different script right?